roles/openshift_repos/tasks/centos_repos.yml

```diff
@@ -10,15 +10,20 @@
     dest: "/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-PaaS"
   notify: refresh cache
 
-# openshift_version is formatted to a standard string in openshift_version role.
-# openshift_version is expected to be in format 'x.y.z...' here.
-# Here, we drop the '.' characters and try to match the correct repo template
-# for our corresponding openshift_version.
+# The version is hardcoded since we cannot use the openshift_version role
+# (it uses the installed packages to determine the version).
 - name: Configure correct origin release repository
   template:
-    src: "{{ item }}"
-    dest: "/etc/yum.repos.d/{{ (item | basename | splitext)[0] }}"
-  with_first_found:
-    - "CentOS-OpenShift-Origin{{ ((openshift_version | default('')).split('.')[0:2] | join('')) }}.repo.j2"
-    - "CentOS-OpenShift-Origin.repo.j2"
+    src: "CentOS-OpenShift-Origin310.repo.j2"
+    dest: "/etc/yum.repos.d/CentOS-OpenShift-Origin310.repo"
   notify: refresh cache
+
+# In some cases, the "latest" repository was used instead of the release repo
+# due to https://github.com/openshift/openshift-ansible/issues/10445.
+# Remove the repository to prevent issues when upgrading, now that we deploy
+# the correct one.
+- name: Remove CentOS-OpenShift-Origin.repo
```

Comment:
I don't think its a good idea to automatically remove something. Admin should be able to remove those with a single ansible command if the systems are affected by #10445